### PR TITLE
[Enhancement] optimize iceberg const column

### PR DESCRIPTION
**Pull request by zombee0**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 20 / 32 (62.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/connector/hive_connector.cpp|11|20|55.00%|[187, 188, 205, 208, 211, 228, 229, 231, 232]|
|:large_blue_circle:|src/runtime/descriptors.cpp|4|7|57.14%|[593, 594, 595]|
|:large_blue_circle:|src/formats/parquet/file_reader.cpp|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 18 / 61 (29.51%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|0|1|00.00%|[1155]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|1|10|10.00%|[141, 222, 223, 224, 225, 226, 227, 228, 229]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|15|48|31.25%|[170, 173, 174, 175, 183, 184, 185, 186, 187, 188, 189, 192, 193, 194, 196, 197, 199, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 256, 259, 260, 276, 277]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 14 / 39 (35.90%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/runtime/descriptors.cpp|0|9|00.00%|[151, 152, 153, 185, 186, 187, 593, 594, 595]|
|:large_blue_circle:|src/connector/hive_connector.cpp|7|23|30.43%|[181, 182, 183, 184, 185, 188, 204, 205, 206, 207, 209, 228, 229, 230, 231, 232]|
|:large_blue_circle:|src/formats/parquet/file_reader.cpp|7|7|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 14 / 39 (35.90%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/runtime/descriptors.cpp|0|9|00.00%|[151, 152, 153, 185, 186, 187, 593, 594, 595]|
|:large_blue_circle:|src/connector/hive_connector.cpp|7|23|30.43%|[181, 182, 183, 184, 185, 188, 204, 205, 206, 207, 209, 228, 229, 230, 231, 232]|
|:large_blue_circle:|src/formats/parquet/file_reader.cpp|7|7|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 60 / 61 (98.36%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|47|48|97.92%|[170]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|1|1|100.00%|[]|

**Review by DorianZheng — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

File: be/src/connector/hive_connector.cpp [hunk 1]
```diff
@@ -177,14 +177,39 @@ Status HiveDataSource::_init_partition_values() {
...
+        }
+
         if (!partition_chunk->has_rows()) {
             _filter_by_eval_partition_conjuncts = true;
         }
     }
     return Status::OK();
 }
 
+int32_t HiveDataSource::is_scan_range_indicate_const_column(SlotId id) const {
```
Comment (dirtysalt) [inline be/src/connector/hive_connector.cpp:200]: return index or bool? 

if index, it's not good name for `is_scan_xxx`.

File: be/src/formats/parquet/file_reader.cpp [hunk 2]
```diff
@@ -503,7 +503,16 @@ Status FileReader::_init_group_readers() {
             auto row_group_reader =
                     std::make_shared<GroupReader>(_group_reader_param, i, _need_skip_rowids, row_group_first_row);
             _row_group_readers.emplace_back(row_group_reader);
-            _total_row_count += _file_metadata->t_metadata().row_groups[i].num_rows;
+            int64_t num_rows = _file_metadata->t_metadata().row_groups[i].num_rows;
+            // for iceberg v2 pos delete
```
Comment (dirtysalt) [inline be/src/formats/parquet/file_reader.cpp:507]: what's wrong if we don't handle delete?  wrong total_row_count?

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

File: be/src/formats/parquet/file_reader.cpp (hunk 2 above)
Comment (zombee0) [reply be/src/formats/parquet/file_reader.cpp:507]: yes wrong total_row_count

File: be/src/connector/hive_connector.cpp (hunk 1 above)
Comment (zombee0) [reply be/src/connector/hive_connector.cpp:200]: yes, it will be better.

**Pull request merged**